modules/assim.sequential/R/Analysis_sda.R

```diff
@@ -272,34 +271,123 @@ GEF<-function(setting,Forecast,Observed, H, extraArg, nitr=50000, nburnin=10000,
+  }else{
+    X_pft2total_start <- 0
+    X_pft2total_end <- 0
+    X_pft2total_model <- 0
+    pft2total_TRUE = FALSE
+  }
+  
+  
+  if(t > 1 & recompile != TRUE){
+    if(X_direct_start != constants.tobit$X_direct_start) recompile = TRUE
+    if(X_direct_end != constants.tobit$X_direct_end) recompile = TRUE
+    if(X_fcomp_end != constants.tobit$X_fcomp_end) recompile = TRUE
+    if(X_fcomp_start != constants.tobit$X_fcomp_start) recompile = TRUE
+  }
+  
+  if(t == 1 | recompile){ #TO DO need to make something that works to pick whether to compile or not
+    constants.tobit <<- list(
+      N = ncol(X),
+      YN = length(y.ind),
+      X_direct_start = X_direct_start,
+      X_direct_end = X_direct_end,
+      X_fcomp_start = X_fcomp_start,
+      X_fcomp_end = X_fcomp_end, #TO DO: FIX
+      X_fcomp_model_start = X_fcomp_model[1],
+      X_fcomp_model_end = X_fcomp_model[length(X_fcomp_model)],
+      X_pft2total_start = X_pft2total_start,
+      X_pft2total_model_start = X_pft2total_model[1],
+      X_pft2total_model_end = X_pft2total_model[length(X_pft2total_model)],
+      direct_TRUE = direct_TRUE,
+      fcomp_TRUE = fcomp_TRUE,
+      pft2total_TRUE = pft2total_TRUE
+    )
     
     dimensions.tobit = list(X = length(mu.f), X.mod = ncol(X),
-                            Q = c(length(mu.f),length(mu.f)))
-    # Data need to be used in the model
+                            Q = c(length(mu.f),length(mu.f)),
+                            y_star = (length(y.censored)))
+    
     data.tobit = list(muf = as.vector(mu.f),
                       pf = solve(Pf), 
                       aq = aqq[t,,], bq = bqq[t],
                       y.ind = y.ind,
                       y.censored = y.censored,
-                      r = solve(R))
-    #initial values
-    inits.pred = list(q = diag(length(mu.f)),
-                      X.mod = as.vector(mu.f),
-                      X = as.vector(mu.f) # This was this before rnorm(length(mu.f),0,1), I thought the mu.f would be a better IC for something like abv ground biomass than something close to zero.
-                      ) #
+                      r = R) #precision
+    
+    inits.pred = list(q = diag(length(mu.f))*(length(mu.f)+1), X.mod = as.vector(mu.f),
+                      X = rnorm(length(mu.f),mu.f,1),
+                      y_star = rnorm(length(y.censored),0,1)) #
     
     model_pred <- nimbleModel(tobit.model, data = data.tobit, dimensions = dimensions.tobit,
                               constants = constants.tobit, inits = inits.pred,
                               name = 'base')
+    
+    model_pred$initializeInfo()
     ## Adding X.mod,q,r as data for building model.
     conf <- configureMCMC(model_pred, print=TRUE)
-    conf$addMonitors(c("X","q","Q")) 
+    conf$addMonitors(c("X","q","Q", "y_star","y.censored")) 
     ## [1] conjugate_dmnorm_dmnorm sampler: X[1:5]
+    
+    
+    if(FALSE){ ### Need this for when the state variables are on different scales like NPP and AGB
```

Comment:
if this needs to be run under certain conditions, then it can't live in a FALSE block. Needs a flag and documentation for what this does and when you need to use it

Reply:
👍

Reply:
Could you also explain what are you doing here ? or why are you changing the sampler ?